### ISSUE TITLE
[FEATURE] Improve type annotation (and documentation?) for `TriangleMesh.at` indexing methods

### ISSUE THREAD

**Issue by jeertmans** [jeertmans/DiffeRT#364]
### Terms

- [x] Checked the [existing issues](https://github.com/jeertmans/DiffeRT/issues?q=is%3Aissue+label%3Aenhancement) to see if my suggestion has not already been suggested;

### Description

Currently, the `TriangleMesh.at` methods, such as `add`, do not provide information about the accepted keyword arguments.

I'm not sure if it's feasible or worth the hassle, but wouldn't it be nice to automatically derive allowed keyword arguments (and maybe their documentation) from `jax.numpy.ndarray.at.get`?

### Screenshots

<img width="575" height="89" alt="Image" src="https://github.com/user-attachments/assets/fca284ef-ccbb-4af8-b013-af6e47862cbc" />

<img width="575" height="89" alt="Image" src="https://github.com/user-attachments/assets/f8783881-26db-444e-b93d-532a31144846" />

### Additional information

https://docs.jax.dev/en/latest/_autosummary/jax.numpy.ndarray.at.html
